Makefile

```diff
@@ -223,7 +223,7 @@ ENABLE_MIMALLOC ?= 1
 
 # https://github.com/microsoft/mimalloc/issues/512
 # Linking mimalloc via object file on macOS x64 can cause heap corruption
-_MIMALLOC_FILE = libmimalloc.o
+_MIMALLOC_FILE = libmimalloc.a
 _MIMALLOC_INPUT_PATH = CMakeFiles/mimalloc-obj.dir/src/static.c.o
```

Comment:
☝️ There's been some juggling with .o and .a depending on platform:
https://github.com/oven-sh/bun/pull/6800/files (do want more eyes on this)

Reply:
This distinction should be deliberate (but a comment explaining this would be good). It impacts the order malloc gets loaded in. In some platforms we want to keep malloc usable or it will cause strange crashes at runtime when libc functions call free or malloc internally.